OpenRA.Game/Network/OrderManager.cs

```diff
@@ -241,6 +271,51 @@ public void Dispose()
 			disposed = true;
 			Connection?.Dispose();
 		}
+
+		public void TickImmediate()
+		{
+			SendImmediateOrders();
+
+			ReceiveAllOrdersAndCheckSync();
+		}
+
+		public bool TryTick()
+		{
+			var shouldTick = true;
+
+			if (IsNetTick)
+			{
+				// Check whether or not we will be ready for a tick next frame
+				// We don't need to include ourselves in the equation because we can always generate orders this frame
+				shouldTick = pendingPackets.All(p => p.Key == Connection.LocalClientId || p.Value.Count > 0);
+
+				// Send orders only if we are currently ready, this prevents us sending orders too soon if we are
+				// stalling
+				if (shouldTick)
+					SendOrders();
+			}
+
+			var willTick = shouldTick;
+			if (willTick && IsNetTick)
+			{
+				willTick = IsReadyForNextFrame;
+				if (willTick)
+					ProcessOrders();
+			}
+
+			if (willTick)
+				LocalFrameNumber++;
+
+			return willTick;
+		}
+
+		bool IsNetTick => LobbyInfo.GlobalSettings.UseNewNetcode || LocalFrameNumber % Game.NetTickScale == 0;
+
+		public void UpdateLastTickTime(long tick)
```

Comment:
These are the only two places where we use `UseNewNetcode` now and both are trivial so it would be better if we could drop them (and this method - Game can call `orderManager.LastTickTime.AdvanceTickTime(tick);` directly) until the next PR. This also resolves https://github.com/OpenRA/OpenRA/pull/19374/files#r650408781.

When we add this back we should also rename it to `AdvanceTickTime` to match the new helper class.